imod/mf6/auxiliary_variables.py

```diff
@@ -25,11 +25,40 @@ def _get_auxiliary_data_variable_names_mapping(package):
     return result
 
 
-def add_periodic_auxiliary_variable(package):
+def add_periodic_auxiliary_variable(package) -> None:
```

Comment:
due to circular import, the Pakcage class cannot be loaded in this source file, and no type information can be provided on package.

Reply:
This would be a good time to introduce the IPackage interface to break the circular dependency. We already have the IPackageBase in the interface folder.

The interface would be:

```
class IPackage(IPackageBase):
    """
    The base methods and attributes available in all packages
    """

    @property
    @abstractmethod
    def auxiliary_data(self) -> Dict[str, str]:
        raise NotImplementedError
```


In the Package class:

```
class Package(PackageBase, IPackage):

....

@property
def auxiliary_data(self) -> Dict[str, str]:
    if (hasattr(self, "_auxiliary_data")):
        return self._auxiliary_data
    else: 
        return {}
```


In the auxiliarry_variables file:
```
def get_variable_names(package: IPackage):
    ...

```

Reply:
done